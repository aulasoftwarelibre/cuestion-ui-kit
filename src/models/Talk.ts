export interface Talk {
  avatar?: string;
  description: string;
  id: string;
  room: string;
  title: string;
  startsAt: Date;
  endsAt: Date;
  speaker: string;
  topics: string[];
}

export default Talk;
