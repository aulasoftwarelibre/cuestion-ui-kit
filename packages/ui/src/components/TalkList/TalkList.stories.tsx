import { action } from "@storybook/addon-actions";
import { array, object, text, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { talks } from "../../models/Talk.mock";
import { wInfo } from "../../utils/wInfo";
import TalkList from "./TalkList";

const stories = storiesOf("Components/TalkList", module);
stories.addDecorator(withKnobs);

stories
  .add(
    "default",
    wInfo()(() => (
      <TalkList
        handleOnClick={action("talk-item")}
        filter={[]}
        talks={object("Talks", talks)}
        title={text("title", "Talk list")}
      />
    )),
  )
  .add(
    "filtered",
    wInfo()(() => (
      <TalkList
        handleOnClick={action("talk-item")}
        filter={array("Filter", ["Agile"])}
        talks={object("Talks", talks)}
        title={text("title", "Talk list")}
      />
    )),
  );
