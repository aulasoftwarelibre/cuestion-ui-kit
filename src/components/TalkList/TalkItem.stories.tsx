import { action } from "@storybook/addon-actions";
import { object, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";

import { Talk } from "../../models/Talk";
import { wInfo } from "../../utils/wInfo";
import TalkItemWrapper from "./TalkItem";

const stories = storiesOf("Components/TalkItem", module);
stories.addDecorator(withKnobs);

const talk: Talk = {
    avatar: "https://i.pravatar.cc/150?img=22",
// tslint:disable-next-line: max-line-length
    description: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed accumsan quis neque at dignissim. Donec auctor bibendum quam sed elementum. Sed gravida, arcu porta sollicitudin porttitor, nisl ligula placerat massa, eu fermentum magna sapien eu sem. Nulla facilisi. Aliquam erat volutpat",
    endsAt: new Date(),
    id: "1",
    room: "Room 1",
    speaker: "John Doe",
    startsAt: new Date("2019/09/13 09:30:00 GMT+2"),
    title: "My awesome talk",
    topics: ["Mobile", "Web"]
};

stories.add(
    "default",
    wInfo()(() => (
        <TalkItemWrapper
            handleOnClick={action("talk-item")}
            talk={object("Talk", talk)}
        />
    ))
);
