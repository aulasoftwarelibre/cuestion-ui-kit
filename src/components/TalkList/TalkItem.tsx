import Avatar from "@material-ui/core/Avatar";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardActions from "@material-ui/core/CardActions";
import CardContent from "@material-ui/core/CardContent";
import CardHeader from "@material-ui/core/CardHeader";
import Chip from "@material-ui/core/Chip";
import Collapse from "@material-ui/core/Collapse";
import Divider from "@material-ui/core/Divider";
import IconButton from "@material-ui/core/IconButton";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";
import Alarm from "@material-ui/icons/Alarm";
import ExpandMore from "@material-ui/icons/ExpandMore";
import Room from "@material-ui/icons/Room";

import clsx from "clsx";
import * as React from "react";
import { InjectedIntl, injectIntl } from "react-intl";

import { Talk } from "../../models/Talk";

export interface Props {
  handleOnClick: any;
  intl: InjectedIntl;
  talk: Talk;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    avatar: {
      backgroundColor: "#60102f"
    },
    card: {
      maxWidth: 345
    },
    chip: {
      marginRight: theme.spacing(0.5)
    },
    expand: {
      marginLeft: "auto",
      transform: "rotate(0deg)",
      transition: theme.transitions.create("transform", {
        duration: theme.transitions.duration.shortest
      })
    },
    expandOpen: {
      transform: "rotate(180deg)"
    },
    media: {
      height: 140
    }
  })
);

const TalkItemWrapper: React.FunctionComponent<Props> = ({
  handleOnClick,
  intl,
  talk
}) => {
  const classes = useStyles();
  const [expanded, setExpanded] = React.useState(false);

  function handleExpandClick() {
    setExpanded(!expanded);
  }

  const startHour = intl.formatTime(talk.startsAt);

  const topics = talk.topics.map(topic => (
    <Chip key={`${talk.id}-${topic}`} label={topic} className={classes.chip} />
  ));

  return (
    <Card className={classes.card}>
      <CardActionArea onClick={() => handleOnClick(talk.id)}>
        <CardHeader
          avatar={<Avatar src={talk.avatar} />}
          title={talk.title}
          subheader={talk.speaker}
        />
        <CardContent>{topics}</CardContent>
      </CardActionArea>
      <Divider variant="middle" />

      <CardActions>
        <Chip
          icon={<Room />}
          label={talk.room}
          size="small"
          variant="outlined"
        />
        <Chip
          icon={<Alarm />}
          label={startHour}
          size="small"
          variant="outlined"
        />
        <IconButton
          className={clsx(classes.expand, {
            [classes.expandOpen]: expanded
          })}
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="Show more"
        >
          <ExpandMore />
        </IconButton>
      </CardActions>
      <Collapse in={expanded} timeout="auto" unmountOnExit={true}>
        <CardContent>
          <Typography paragraph={true} variant="body2">
            {talk.description}
          </Typography>
        </CardContent>
      </Collapse>
    </Card>
  );
};

export const TalkItem = injectIntl(TalkItemWrapper);

export default TalkItem;
