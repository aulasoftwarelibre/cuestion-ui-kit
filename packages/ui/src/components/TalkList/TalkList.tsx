import * as React from "react";

import { Grid, Typography } from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import Talk from "../../models/Talk";
import TalkItem from "../TalkItem/TalkItem";

export interface Props {
  handleOnClick: any;
  filter: string[];
  talks: Talk[];
  title: string;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      overflow: "hidden",
      padding: theme.spacing(0, 1)
    },
    grid: {
      padding: theme.spacing(2)
    }
  })
);

export const TalkList: React.FunctionComponent<Props> = ({ handleOnClick, filter, talks, title }) => {
  const classes = useStyles();

  const list = talks.map(talk => (
    <Grid item xs={12} justify="flex-start" md={4} key={talk.id} className={classes.grid}>
      <TalkItem handleOnClick={handleOnClick} talk={talk} raised={filter.some(r => talk.topics.includes(r))} />
    </Grid>
  ));

  return (
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid item key="title" xs={12}>
          <Typography component="h2" variant="h6" color="primary">
            {title}
          </Typography>
        </Grid>
        {list}
      </Grid>
    </div>
  );
};

export default TalkList;
