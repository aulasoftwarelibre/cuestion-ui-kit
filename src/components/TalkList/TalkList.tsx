import * as React from "react";

import { Grid, Typography } from "@material-ui/core";
import Talk from "../../models/Talk";
import TalkItem from "../TalkItem/TalkItem";

export interface Props {
  handleOnClick: any;
  filter: string[];
  talks: Talk[];
  title: string;
}

export const TalkList: React.FunctionComponent<Props> = ({ handleOnClick, filter, talks, title }) => {
  const list = talks.map(talk => (
    <Grid item xs={6} md={4} key={talk.id}>
      <TalkItem handleOnClick={handleOnClick} talk={talk} raised={filter.some(r => talk.topics.includes(r))} />
    </Grid>
  ));
  return (
    <div>
      <Grid container spacing={2}>
        <Grid item key="title" xs={12}>
          <Typography component="h2" variant="h6" color="primary">
            {title}
          </Typography>
        </Grid>
        {list}
      </Grid>
    </div>
  );
};

export default TalkList;
