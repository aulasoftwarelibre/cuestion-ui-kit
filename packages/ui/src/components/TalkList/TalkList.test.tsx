import * as React from "react";

import CardActionArea from "@material-ui/core/CardActionArea";
import { talks } from "../../models/Talk.mock";
import {
  createComponentWithIntl,
  mountWithIntl,
} from "../../utils/createComponentWithIntl";
import TalkList from "./TalkList";

describe("TalkList", () => {
  const mockOnHandleClick = jest.fn();

  it("should render correctly without filter", () => {
    const component = createComponentWithIntl(
      <TalkList
        handleOnClick={mockOnHandleClick}
        filter={[]}
        talks={talks}
        title={"Talk List"}
      />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with filter", () => {
    const component = createComponentWithIntl(
      <TalkList
        handleOnClick={mockOnHandleClick}
        filter={["Agile"]}
        talks={talks}
        title={"Talk List"}
      />,
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it("should return message from talk selected", () => {
    const wrapper = mountWithIntl(
      <TalkList
        handleOnClick={mockOnHandleClick}
        filter={[]}
        talks={talks}
        title={"Talk List"}
      />,
    );

    wrapper
      .find(CardActionArea)
      .first()
      .simulate("click");

    expect(mockOnHandleClick).toBeCalledWith(talks[0].id);
  });
});
