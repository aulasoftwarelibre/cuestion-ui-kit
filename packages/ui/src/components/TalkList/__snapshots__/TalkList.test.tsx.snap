// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TalkList should render correctly with filter 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
      >
        Talk List
      </h2>
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root"
              >
                <img
                  className="MuiAvatar-img"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/jydesign/128.jpg"
                />
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                My awesome talk
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Larue Johnston
              </span>
            </div>
          </div>
          <div
            className="MuiCardContent-root"
          >
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Mobile
              </span>
            </div>
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Web
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              Room 1
            </span>
          </div>
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              9:30 AM
            </span>
          </div>
          <button
            aria-expanded={false}
            aria-label="Show more"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-138"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation8 MuiCard-root MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root"
              >
                <img
                  className="MuiAvatar-img"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/thibaut_re/128.jpg"
                />
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                My amazing talk
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Brian Kuphal
              </span>
            </div>
          </div>
          <div
            className="MuiCardContent-root"
          >
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Frontend
              </span>
            </div>
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Agile
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              Room 2
            </span>
          </div>
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              9:30 AM
            </span>
          </div>
          <button
            aria-expanded={false}
            aria-label="Show more"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-138"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`TalkList should render correctly without filter 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
    >
      <h2
        className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
      >
        Talk List
      </h2>
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root"
              >
                <img
                  className="MuiAvatar-img"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/jydesign/128.jpg"
                />
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                My awesome talk
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Larue Johnston
              </span>
            </div>
          </div>
          <div
            className="MuiCardContent-root"
          >
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Mobile
              </span>
            </div>
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Web
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              Room 1
            </span>
          </div>
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              9:30 AM
            </span>
          </div>
          <button
            aria-expanded={false}
            aria-label="Show more"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-138"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-4"
    >
      <div
        className="MuiPaper-root MuiPaper-elevation1 MuiCard-root MuiPaper-rounded"
      >
        <button
          className="MuiButtonBase-root MuiCardActionArea-root"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="MuiCardHeader-root"
          >
            <div
              className="MuiCardHeader-avatar"
            >
              <div
                className="MuiAvatar-root"
              >
                <img
                  className="MuiAvatar-img"
                  src="https://s3.amazonaws.com/uifaces/faces/twitter/thibaut_re/128.jpg"
                />
              </div>
            </div>
            <div
              className="MuiCardHeader-content"
            >
              <span
                className="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
              >
                My amazing talk
              </span>
              <span
                className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
              >
                Brian Kuphal
              </span>
            </div>
          </div>
          <div
            className="MuiCardContent-root"
          >
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Frontend
              </span>
            </div>
            <div
              className="MuiChip-root makeStyles-chip-137"
              onKeyDown={[Function]}
              onKeyUp={[Function]}
            >
              <span
                className="MuiChip-label"
              >
                Agile
              </span>
            </div>
          </div>
          <span
            className="MuiCardActionArea-focusHighlight"
          />
        </button>
        <hr
          className="MuiDivider-root MuiDivider-middle"
        />
        <div
          className="MuiCardActions-root MuiCardActions-spacing"
        >
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7zm0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              Room 2
            </span>
          </div>
          <div
            className="MuiChip-root MuiChip-outlined MuiChip-sizeSmall"
            onKeyDown={[Function]}
            onKeyUp={[Function]}
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiChip-icon MuiChip-iconSmall"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 5.72l-4.6-3.86-1.29 1.53 4.6 3.86L22 5.72zM7.88 3.39L6.6 1.86 2 5.71l1.29 1.53 4.59-3.85zM12.5 8H11v6l4.75 2.85.75-1.23-4-2.37V8zM12 4c-4.97 0-9 4.03-9 9s4.02 9 9 9c4.97 0 9-4.03 9-9s-4.03-9-9-9zm0 16c-3.87 0-7-3.13-7-7s3.13-7 7-7 7 3.13 7 7-3.13 7-7 7z"
              />
            </svg>
            <span
              className="MuiChip-label MuiChip-labelSmall"
            >
              9:30 AM
            </span>
          </div>
          <button
            aria-expanded={false}
            aria-label="Show more"
            className="MuiButtonBase-root MuiIconButton-root makeStyles-expand-138"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
